internal/factsengine/gatherers/saplocalhostresolver_test.go

```diff
@@ -0,0 +1,95 @@
+package gatherers_test
+
+import (
+	"io"
+	"os"
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/suite"
+	"github.com/trento-project/agent/internal/factsengine/gatherers"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	utilsMocks "github.com/trento-project/agent/pkg/utils/mocks"
+	"github.com/trento-project/agent/test/helpers"
+)
+
+type SapLocalhostResolverTestSuite struct {
+	suite.Suite
+	mockResolver *utilsMocks.HostnameResolver
+}
+
+func TestSapLocalhostResolverTestSuite(t *testing.T) {
+	suite.Run(t, new(SapLocalhostResolverTestSuite))
+}
+
+func (suite *SapLocalhostResolverTestSuite) SetupTest() {
+	suite.mockResolver = new(utilsMocks.HostnameResolver)
+}
+
+func (suite *SapLocalhostResolverTestSuite) TestSapLocalhostResolverSuccess() {
+	appFS := afero.NewMemMapFs()
+
+	err := appFS.MkdirAll("/usr/sap/PRD", 0644)
+	suite.NoError(err)
+	err = appFS.MkdirAll("/usr/sap/QAS", 0644)
+	suite.NoError(err)
+
+	defaultProfileFile, _ := os.Open(helpers.GetFixturePath("gatherers/sap_profile.default"))
+	defaultProfileContent, _ := io.ReadAll(defaultProfileFile)
+	err = afero.WriteFile(appFS, "/sapmnt/PRD/profile/DEFAULT.PFL", defaultProfileContent, 0644)
+	suite.NoError(err)
+	err = afero.WriteFile(appFS, "/sapmnt/PRD/profile/DEFAULT.1.PFL", []byte{}, 0644)
```

Comment:
We don't need this now. The `FindProfiles` is already tested and we know that it excludes the backups